src/olympia/addons/serializers.py

```diff
@@ -1,33 +1,120 @@
+from olympia.addons.models import Addon, attach_tags
+from olympia.amo.helpers import absolutify
 from olympia.api.fields import TranslationSerializerField
 from olympia.api.serializers import BaseESSerializer
+from olympia.files.models import File
+from olympia.versions.models import Version
+
+
+class FileSerializer(serializers.ModelSerializer):
+    url = serializers.SerializerMethodField('get_url')
+    platform = serializers.Field(source='get_platform_display')
+    status = serializers.Field(source='get_status_display')
+
+    class Meta:
+        model = File
+        fields = ('id', 'created', 'hash', 'platform', 'size', 'status', 'url')
+
+    def get_url(self, obj):
+        return obj.get_url_path(src='')
+
+
+class VersionSerializer(serializers.ModelSerializer):
+    files = FileSerializer(source='all_files', many=True)
+
+    # FIXME:
+    # - license
+    # - appversion compatibility info
+    # - release notes (separate endpoint ?)
+    # - all the reviewer/admin fields (different serializer/endpoint)
+
+    class Meta:
+        model = Version
+        fields = ('id', 'files', 'reviewed', 'version')
 
 
 class AddonSerializer(serializers.ModelSerializer):
-    name = TranslationSerializerField()
+    current_version = VersionSerializer()
     description = TranslationSerializerField()
+    homepage = TranslationSerializerField()
+    name = TranslationSerializerField()
+    status = serializers.Field(source='get_status_display')
+    summary = TranslationSerializerField()
+    support_email = TranslationSerializerField()
+    support_url = TranslationSerializerField()
+    tags = serializers.SerializerMethodField('get_tags')
+    type = serializers.Field(source='get_type_display')
+    url = serializers.SerializerMethodField('get_url')
+
+    # FIXME:
+    # - categories (need to sort out the id/slug mess in existing search code)
+    # - icon/previews
+    # - average rating, number of downloads, hotness
+    # - dictionary-specific things
+    # - persona-specific things
+    # - contributions-related things
+    # - annoying/thankyou and related fields
+    # - authors
+    # - dependencies, site_specific, external_software
+    # - thereason/thefuture (different endpoint ?)
+    # - in collections, other add-ons by author, eula, privacy policy
+    # - eula / privacy policy (different endpoint)
+    # - all the reviewer/admin-specific fields (different serializer/endpoint)
 
     class Meta:
         model = Addon
-        fields = ('id', 'default_locale', 'name', 'last_updated', 'slug')
+        fields = ('id', 'current_version', 'default_locale', 'description',
+                  'guid', 'homepage', 'name', 'last_updated', 'public_stats',
+                  'slug', 'status', 'summary', 'support_email', 'support_url',
+                  'tags', 'type', 'url')
+
+    def get_tags(self, obj):
+        if not hasattr(obj, 'tag_list'):
+            attach_tags([obj])
+        return getattr(obj, 'tag_list', [])
```

Comment:
Actually I do, `attach_tags` does not do anything if there are no tags to attach :( I'll add a comment.